pkg/intermediate/aggregate.go

```diff
@@ -984,3 +995,29 @@ func isCorrelationRequired(flowType uint8, record entities.Record) bool {
 	}
 	return false
 }
+
+func fillHttpVals(incomingHttpVals, existingHttpVals string) (string, error) {
+	incomingHttpValsJson := make(map[int32]string)
+	existingHttpValsJson := make(map[int32]string)
+
+	if incomingHttpVals != "" {
+		if err := json.Unmarshal([]byte(incomingHttpVals), &incomingHttpValsJson); err != nil {
+			return "", fmt.Errorf("error parsing JSON: %v", err)
+		}
+	}
+	if existingHttpVals != "" {
+		if err := json.Unmarshal([]byte(existingHttpVals), &existingHttpValsJson); err != nil {
+			return "", fmt.Errorf("error parsing JSON: %v", err)
+		}
+	}
+	if len(existingHttpValsJson) > 0 {
```

Comment:
the check is not useful: if the length is 0, the the loop will simply not execute